src/static/js/modules/loan-comparison/stores/loan-store.js

```diff
@@ -37,34 +37,57 @@ var LoanStore = assign({}, EventEmitter.prototype, {
         if (this._loans.length == 0 || scenario) {
             this.updateDownpaymentConstant(scenario);
             for (i = 0; i < len; i++) {
-                this.resetLoan(i, scenario);
+                this._loans[i] = this.resetLoan(i, this._loans[i], scenario);
             }
         }    
     },
     
     /**
-     * Resets a loan's properties using a combination of default loan data,
+     * Resets a loan's properties.
+     * Also runs methods to update dependencies/validations/calculations,
+     * and initiates api requests.
+     *
+     * @param  {number} id 
+     * @param  {object} loan 
+     * @param  {object} scenario optional
+     * @return  {object} loan
+     */
+     resetLoan: function (id, loan, scenario) {
+         loan = LoanStore.setupLoanData(id, loan || {}, scenario);
+         
+         // ensure downpayment percent because default data only has downpayment
+         if (!loan['downpayment-percent']) {
+             LoanStore.updateCalculatedValues(loan, 'downpayment-percent');
+         }
+         LoanStore.updateCalculatedValues(loan, ['loan-amount', 'loan-summary']);
+         LoanStore.validateLoan(loan);
+         LoanStore.fetchLoanData(loan);
+         // TODO: add county request as option to fetchLoanData?
+         LoanStore.fetchCounties(loan, true);
+         
+         return loan;
+     },
+    
+    /**
+     * Sets up a loan object using a combination of default loan data,
      * existing loan data, and data specific to the current scenario.
      * If there's a scenario, loans will have the same values for 
      * all properties except scenario-specific ones, so we use the 
-     * first loan's data as existingData for all loans (but omit id & api requests).
-     * Also runs updateLoan to update dependencies/validations/calculations,
-     * and initiates api requests.
+     * first loan's data as existingData for loan 2+ (omitting id & api requests)
      *
      * @param  {number} id 
+     * @param  {object} loan 
      * @param  {object} scenario optional
+     * @return  {object} loan
      */
-    resetLoan: function (id, scenario) {
+    setupLoanData: function (id, loan, scenario) {
+        var defaultData = common.defaultLoanData;
         var scenarioData = scenario ? scenario.loanProps[id] : {};        
-        var existingData = this._loans[id];
-        
-        if (scenario && id > 0) {
-            existingData = common.omit(this._loans[0], 'id', 'rate-request', 'mtg-ins-request');
-        }
+        var existingData = scenario && id > 0 
```

Comment:
There would be existingData when transitioning into custom mode too. The main difference is that in scenarios, the first loan's data serves as existingData for loan 2 as well.

This whole function should go away when we remove the prepop scenarios, though. :)